### DISCUSSION TITLE
Integrating OceanBioME with data assimilation package

### DISCUSSION THREAD

**Discussion by rj678** [OceanBioME/OceanBioME.jl#163] [General]
Hello,

 Thank you for releasing this amazing repo!

 In your paper, you mention that "implementation of OceanBioME.jl models allows for seamless integration with data
assimilation packages, such as EnsembleKalmanProcesses.jl" 

 It'd help me greatly if you could share an example where you integrate OceanBioMe with EnsembleKalmanProcesses

Thank you!

**Comment by navidcy** [comment]
hi @rj678, do you need an example with OceanBioME.jl + EnsembleKalmanProcesses.jl? Because if not, we do have examples with just Oceananigans.jl + EnsembleKalmanProcesses.jl I can point you to.

**Comment by rj678** [comment]
thank you for replying - @navidcy , that's right - I was looking for an example of the "seamless integration" of OceanBioME.jl + EnsembleKalmanProcesses.jl - I have gone through the individual examples of the 2 libraries

**Comment by jagoosw** [comment]
Hi @rj678,

To use OceanBioME + EnsembleKalmanProcesses is almost identical to using it with Oceananigans but I'll put an example together in the next few days. In the meantime for my models which don't need every ensemble member to be executed on different nodes I usually just use the first example in the EnsembleKalmanProcesses docs and change `G(u)` to be whatever model I need.

The main reason OceanBioME is easily integrated with a data assimilation package (and the same goes for Oceananigans) is because of how the model components are formulated as objects which can be specified with different parameter values. This differs from "traditional" models where you have to make setup scripts (e.g. https://zenodo.org/records/7139521) which is less conducive to easy data assimilation.

**Comment by rj678** [comment]
thank you so much for the detailed response @jagoosw and for sharing an example of a comparable traditional model - I look forward to the OceanBioME + EnsembleKalmanProcesses example and I'm very excited to learn more about the functionality of OceanBioME

**Comment by jagoosw** [comment]
I've put an example here https://github.com/OceanBioME/OceanBioME.jl/pull/164 where we learn a few of the NPZD model parameters.

Hopefully, it is clear enough how you can adapt it for real cases. The EKP docs examples will hopefully be more helpful in showing how to scale it for larger simulations etc.

**Comment by rj678** [comment]
amazing - that's super helpful - thank you so much!
